lib/uklibparam/parser.c

```diff
@@ -597,12 +566,11 @@ static int parse_arg(struct parse_arg_ctx *ctx, char *strbuf, int scan_only)
 			 *       not allowed characters, or empty strings).
 			 */
 			if (!scan_only) {
-				ctx->ld = find_libdesc(libname, libname_len);
-				if (ctx->ld)
-					ctx->p = find_libparam(ctx->ld,
-							       paramname,
-							       paramname_len);
-				if (!ctx->ld || !ctx->p) {
+				ctx->p = find_libparam(libname,
```

Comment:
Hum, this changes the potential number of required comparisons. A library with many parameters will cause that the library name is compared much more often. It will be costly for arguments that belong to a library with a big list of parameters. However, because we need to see first how much this in reality impacts the runtime performance, I would add a comment for now that briefly mentions that we could optimize this in the future with an adopted layout if needed.

Reply:
Yes I noticed that too, but I believe that in practice the possibility to come across such a case is slim to be worth implementing a more sophisticated scheme. I'll add a comment.

Reply:
Thanks, and yes, I agree with you.

Reply:
I am thinking of a `TODO` comment instead of a `FIXME` because functionality-wise, everything is correct. There is also no shortcut/simplification that ignores corner cases.

I thought along these lines:
```c
/*
 * TODO: This can become expensive if the number of parameters for a
 *       library becomes high: The library name is repeatedly checked
 *       for a match for each library parameter (O(n2)). However, the
 *       impact in practice has yet to be assessed.
 */